### PR TITLE
Api endpoint

### PR DESCRIPTION
**Pull request by mmsesay**
In this PR we worked on the following:

- Create an API endpoint to list all posts.
- Create an API endpoint to list all comments for a post.
- Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter.
- Add authentication to your API endpoints.
- Your API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

File: app/controllers/api/v1/application_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module Api
+  module V1
+    class ApplicationController < ActionController::API
+      include ExceptionHandler
+
+      # def json_payload
+      #   HashWithIndifferentAccess.new(JSON.parse(request.raw_post))
+      # end
```
Comment (arslanbisharat) [inline app/controllers/api/v1/application_controller.rb]: - [ ] Let's remove the commented code from your project.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by DeliceLydia — APPROVED** [review]
# Congratulations🎉🎉 


## Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Gambit142** [comment]
Thank you for the feedback

**Pull request merged**